header_test.go

```diff
@@ -128,6 +128,31 @@ func TestCompressAllocHdr(t *testing.T) {
 	}
 }
 
+func TestUncompressAllocHdrWithZeroLengthHeader(t *testing.T) {
+	input := []byte("")
+	output := make([]byte, CompressBoundHdr(input))
+	outSize, err := CompressHdr(output, input)
+	if err != nil {
+		t.Fatalf("Compression failed: %v", err)
+	}
+	if outSize == 0 {
+		t.Fatal("Output buffer is empty.")
+	}
+	// This is the key line. Some lz4 implementations don't trim their compressed outputs for blank payloads,
```

Comment:
Oh wow, does this mean we actually have a bug for other inputs also? Like what happens if a non-empty compressed payload has extra data? Or what is UncompressAllocHdr even supposed to do in these cases? :) The doc comment certainly does not specify...

One moment let me test this ...